File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        res = []
        for i in range(len(nums)):
            for j in range(len(nums)):
                for k in range(len(nums)):
                    if i != j and i != k and j != k:
                        current = nums[i] + nums[j] + nums[k]
                        current_vals = [nums[i], nums[j], nums[k]]
                        current_vals.sort()
                        if current == 0 and current_vals not in res:
                            res.append(current_vals)
        return res

File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {

    // O(n * log n) | O(1)
    public boolean hasDuplicate(int[] nums) {
        if (nums.length <= 1) return false;
        
        Arrays.sort(nums);
        int prev = nums[0];

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == prev) return true;
            prev = nums[i];
        }
        return false;
    }
}
